src/server/server_family.cc

```diff
@@ -2222,10 +2225,12 @@ void ServerFamily::Info(CmdArgList args, const CommandContext& cmd_cntx) {
     append("dragonfly_version", GetVersion());
     append("redis_mode", GetRedisMode());
     append("arch_bits", 64);
-    append("os", GetOSString());
+    if (!absl::GetFlag(FLAGS_managed_service_info)) {
```

Comment:
We could cache the result of GetFlag :)

Reply:
We shouldn't optimize for `INFO` command in that level, let alone this value can change via `CONFIG SET`.

Reply:
If it can change by `CONFIG SET` then an `INFO ALL` might contain `some` of the changes depending on the `GetFlag` at that point in time. In other words this is a race (not a data race but a race on what the thread will see when it polls GetFlag twice)

Reply:
We can surely live our lives with this race :rofl: